openpype/plugins/publish/extract_slate_global.py

```diff
@@ -0,0 +1,541 @@
+        common_data["comment"] = instance.context.data.get("comment") or "-"
+        intent = instance.context.data.get("intent")
+        if not isinstance(intent, dict):
+            intent = {
+                "label": intent,
+                "value": intent
+            }
+        common_data["intent"] = intent
+
+        # init SlateCreator Object that creates the slate frames
+        slate_creator = SlateCreator(
+            template_path=slate_data["slate_template_path"],
+            resources_path=slate_data["slate_resources_path"],
+            data=common_data,
+        )
+
+        # loop through representations to find thumbnail path
+        repre_thumbnail_path = ""
+        for thumb_repre in instance.data["representations"]:
+            if thumb_repre["name"] == "thumbnail":
+                repre_thumbnail_path = os.path.join(
+                    thumb_repre["stagingDir"], thumb_repre["files"]
+                )
+
+        repre_match = None
+
+        # loop through representations and generate a slate frame for each
+        for repre in instance.data["representations"]:
+            if repre["name"] in repre_ignore_list:
+                self.log.debug(
+                    "Representation '{}' is ignored.".format(repre["name"])
+                )
+                continue
+
+            # check if repre is a sequence
+            is_sequence = isinstance(repre["files"], list)
+            check_file = repre["files"][0] if is_sequence else repre["files"]
+
+            file_path = os.path.normpath(
+                os.path.join(repre["stagingDir"], check_file))
+
+            # if representation is a sequence render out a slate before first
+            # frame
+            if is_sequence:
+                filename, _frame, ext = check_file.split(".")
+                frame_start = int(repre["frameStart"]) - 1
+                frame_end = len(repre["files"]) + frame_start
+                output_name = "{}.{}.{}".format(
+                    filename,
+                    str(frame_start).zfill(int(common_data["frame_padding"])),
+                    ext,
+                )
+                thumbnail_path = os.path.join(
+                    repre["stagingDir"],
+                    "{}_slate_thumb.png".format(repre["name"]),
+                ).replace("\\", "/")
+                slate_creator.render_image_oiio(
+                    file_path.replace("\\", "/"), thumbnail_path
+                )
+                repre_match = instance.data["family"]
+
+            else:  # else find matching tags and transfer
+                frame_start = int(repre["frameStart"])
+                frame_end = int(repre["frameEnd"])
+                thumbnail_path = repre_thumbnail_path
+                output_name = "{}_slate_temp.png".format(repre["name"])
+                for tag in repre["tags"]:
+                    for profile in slate_data["slate_profiles"]:
+                        if tag in profile["families"]:
```

Comment:
Why is a `tag` checked if is in `families`? Those are 2 totally different things...

Reply:
Profiles are usually instance data based, not representation based.